tests/errgroup.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2023 MinIO, Inc.
```

Comment:
will remove this once https://github.com/minio/pkg/pull/63 is merged

will start using `minio/pkg` after that.